nautobot/core/views/routers.py

```diff
@@ -52,6 +52,13 @@ class NautobotUIViewSetRouter(SimpleRouter):
             detail=False,
             initkwargs={"suffix": "Bulk Delete"},
         ),
+        # DynamicRoute that will handle custom actions where detail=False
+        DynamicRoute(
+            url=r"^{prefix}/{url_path}{trailing_slash}$",
```

Comment:
Would be nice to add a custom action to the example-plugin that demonstrates this functionality - would probably be useful in test writing too.

Reply:
sounds good, can you give me some hints where and what the test should look like? Struggling to think about how to go about such a test.

Reply:
A custom view for `ExampleModel` should probably look something like this https://github.com/nautobot/nautobot/compare/u/hanlin-demo?expand=1. However, this does not work there because it seems like predefined static routes are taking precedence over dynamic routes. This is going to `{prefix}/{lookup}` instead of `{prefix}/{url_path}`.
![Screen Shot 2023-09-25 at 11 56 01 PM](https://github.com/nautobot/nautobot/assets/46973263/9fd8a4ce-0178-4530-9cac-1c3d168d23ce)

Also a test case should look similar to this, basically asserting response code 200 and that the content is rendered correctly.
```
def test_list_objects_with_permission(self):
            # Add model-level permission
            obj_perm = users_models.ObjectPermission(name="Test permission", actions=["view"])
            obj_perm.save()
            obj_perm.users.add(self.user)
            obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))

            # Try GET with model-level permission
            response = self.client.get(self._get_url("list"))
            self.assertHttpStatus(response, 200)
            response_body = response.content.decode(response.charset)

            list_url = self.get_list_url()
            title = self.get_title()

            # Check if breadcrumb is rendered correctly
            self.assertIn(
                f'<a href="{list_url}">{title}</a>',
                response_body,
            )

```

Reply:
Are you getting that error when not setting url_path?

Reply:
Yes

Reply:
ok, took me a bit, I didn't realize this wasn't based on my PR. Notice the difference in the order of your dynamicroute vs mine. Once it goes from `details=False` to `details=True` you will run into the issue you are seeing. Move the dynamicroute up one and you should be fine.

Reply:
I'm running into the same issue here with the reordering. https://github.com/nautobot/nautobot/compare/u/hanlin-demo?expand=1.

Reply:
Sorry about the confusion, I added the custom action to the APIViewSet instead of the UIViewSet. Running into a different issue now.

Reply:
Tagging #4512 for traceability.